### ISSUE TITLE
Add support of Tiny C Compiler (TCC) compiler

### ISSUE THREAD

**Issue by nesbox** [nesbox/TIC-80#648]
https://bellard.org/tcc/

**Comment by RobLoach** [comment]
Looks like it's no longer maintained? :cry:

**Comment by Anrock** [comment]
Lots of fresh commits in repo: https://repo.or.cz/w/tinycc.git. Even Fabrice Bellard himself it no longer working on a project - it's opensource and there are a lot of contributors there.

**Comment by nesbox** [comment]
Ok, let's give it a try :)

**Comment by joshgoebel** [comment]
You mean as a language you could use to write cartridges? That would be cool, but...

>  tcc generates x86 code.

Do you have some idea how this would work?  We support running languages in a variety of VMs (Lua, Wren, JS)... so we need something that includes it's own VM or runtime, not native x86 code.

**Comment by joshgoebel** [comment]
https://github.com/jpoirier/picoc (a C interpreter) looks like it might be more up-to-speed, but also seems quite old with no activity?

**Comment by joshgoebel** [comment]
Or C/C++ via WASM, ie #1259.  This should be doable as soon as I push my PR.  Though the workflow might be a bit more complex than if it was "built in" in the same way the other interpreters are.

**Comment by nesbox** [comment]
I wanted to use TCC as a compiler for TIC code, not to write carts :)

**Comment by RobLoach** [comment]
I have to agree with nesbox here. Writing a TIC-80 cart in C when you could chose Lua or Wren seems insane :upside_down_face:

**Comment by joshgoebel** [comment]
Well, C isn't the language I'd pick either, just it happens to be an easy target to support for WASM - so someone who is a super pro with C might feel differently...  :-)

**Comment by RobLoach** [comment]
Ah, I understand the motivation now haha

**Comment by joshgoebel** [comment]
I think porting games from other platforms written in C might be one reason someone would be interested.  I could take my Arduboy Remastered work for WASM-4 and now drop it right into TIC-80 with very little effort because of the ability to compile C and have everything "just work".

**Comment by aliceisjustplaying** [comment]
@nesbox isn't this already supported per https://github.com/nesbox/TIC-80/tree/main/templates/c?

**Comment by joshgoebel** [comment]
Compiled to WASM, yes.

**Comment by aliceisjustplaying** [comment]
ah but I guess you want to be able to write C within TIC-80.

**Comment by joshgoebel** [comment]
TIC-80 includes multiple language VM/runtimes (and may include more in the future) - but actual compilers are a bit out of scope - esp those that generate X86 code.
